src/pages/Register.jsx

```diff
@@ -0,0 +1,163 @@
+import { useRef, useState, useEffect, useReducer } from 'react';
+import { faCheck, faTimes, faInfoCircle } from "@fortawesome/free-solid-svg-icons" ;
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome" ;
+import "./Register.css"
+
+const USER_REGEX = /^[a-zA-Z][a-zA-z0-9-_]{3,23}$/;
+const PWD_REGEX = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%]).{8,24}$/;
+
+function Register () {
+    const userRef = useRef();
+    const errRef = useRef();
+
+    const [user, setUser] = useState('');
+    const [validName, setValidName] = useState(false);
+    const [userFocus, setUserFocus] = useState(false);
+
+    const [pwd, setPwd] = useState('');
+    const [validPwd, setValidPwd] = useState(false);
+    const [pwdFocus, setPwdFocus] = useState(false);
+
+    const [matchPwd, setMatchPwd] = useState('');
+    const [validMatch, setValidMatch] = useState(false);
+    const [matchFocus, setMatchFocus] = useState(false);
+
+    const [errMsg, setErrMsg] = useState('');
+    const [success, setSuccess] = useState(false);
+
+    useEffect(() => {
+        userRef.current.focus();
+    }, [])
+
+    useEffect(() => {
+        setValidName(USER_REGEX.test(user));
+    }, [user])
+
+    useEffect(() => {
+        const isPwdValid = PWD_REGEX.test(pwd);
+        setValidPwd(isPwdValid);
+        setValidMatch(isPwdValid && pwd === matchPwd);
+    }, [pwd, matchPwd])
+
+    useEffect(() => {
+        setErrMsg('');
+    }, [user, pwd, matchPwd])
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        setSuccess(true);
+    }
+
+
+    return (
+        
+        <div className="container mx-auto max-w-3xl px-4 py-8">
+        <h1 className="text-3xl font-bold mb-6">Register</h1>
+
+        {success ? (
+                <section>
+                    <h1>Success!</h1>
+                    <p>
+                        <a href="#">Sign In</a>
+                    </p>
+                </section>
+            ) : (
+                <section>
+                    <p ref={errRef} className={errMsg ? "errmsg" : "offscreen"} aria-live="assertive">{errMsg}</p>    
+                    
+                    <form onSubmit={handleSubmit}>
```

Comment:
Simple style improvement: stack the inputs vertically. Can group each label and input together, then use flexbox with direction of column

Reply:
Done

Reply:
Done